WalletWasabi/Wallets/Wallet.cs

```diff
@@ -39,56 +39,61 @@ public class Wallet : BackgroundService
 	{
 		private Node _localBitcoinCoreNode = null;
 
-		public Wallet(
-			Network network,
+		public Wallet(string dataDir, Network network, string filePath) : this(dataDir, network, KeyManager.FromFile(filePath))
+		{
+		}
+
+		public Wallet(string dataDir, Network network, KeyManager keyManager)
+		{
+			State = WalletState.Uninitialized;
+			DataDir = Guard.NotNullOrEmptyOrWhitespace(nameof(dataDir), dataDir);
+			Network = Guard.NotNull(nameof(network), network);
+			KeyManager = Guard.NotNull(nameof(keyManager), keyManager);
+
+			RuntimeParams.SetDataDir(dataDir);
+			HandleFiltersLock = new AsyncLock();
+
+			BlockFolderLock = new AsyncLock();
+			BlocksFolderPath = Path.Combine(dataDir, "Blocks", Network.ToString());
```

Comment:
nitpick: this is inconsistent. Could you rename one of these two properties? In my opinion we should use singular everywhere for consistence (personally I prefer plural in these cases) so, this should be `BlockFolderPath`.

Reply:
ACK!